Unit tests + npm setup

Dezzles opened the pull request:
Added argument checking on Range constructor
Wrote an initial set of unit tests
```
-------------|----------|----------|----------|----------|-------------------|
File         |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
-------------|----------|----------|----------|----------|-------------------|
All files    |    57.14 |    57.14 |    63.64 |    57.14 |                   |
 quadtree.js |    57.14 |    57.14 |    63.64 |    57.14 |... 52,153,154,157 |
```

shiffman commented on package.json:
```diff
@@ -0,0 +1,37 @@
+    "coding",
+    "train"
+  ],
+  "author": "Dan Shiffman",
```
Coding Train Community?

Dezzles replied:
Updated

shiffman commented:
This is awesome, I'm excited to merge!

Pierstoval commented on quadtree.js:
```diff
@@ -82,9 +82,15 @@ class Circle {
+    }
+    if (capacity < 1) {
+      throw RangeError('capacity must be greater than 0');
+    }
```
These checks are not enough IMO: checking the existence of Boundary still means we can set pears or apples.

To be ideally checked, boundary should be validated as a real boundary element 😄

Pierstoval replied:
Same for "Capacity" by the way

Dezzles replied:
Made these validations more specific, added unit tests for the changes

Pierstoval commented on test/quadtree.spec.js:
```diff
@@ -0,0 +1,183 @@
+    });
+  });
+
+});
```
Missing newline (sorry :p)

Pierstoval commented on test/rectangle.spec.js:
```diff
@@ -0,0 +1,163 @@
+      expect(base.intersects(test)).not.to.be.true;
+    });
+  });
+});
```
Missing newline here too

Dezzles replied:
Good spot on these, fixed

Dezzles commented:
Added additional unit tests to finish out the QuadTree coverage.

One missing branch is related to the below code, so long as nobody has gone ahead and modified the boundaries then the 'else' here should never happened
```
    if (this.northeast.insert(point) || this.northwest.insert(point) ||
      this.southeast.insert(point) || this.southwest.insert(point)) {
      return true;
    }
```

The other comes from the below line where I can't really test for the situation where NodeJS doesn't exist because I'm running my tests from NodeJS
```
((module ? module.exports = { Point, Rectangle, QuadTree, Circle } : 0));
```

```
-------------|----------|----------|----------|----------|-------------------|
File         |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
-------------|----------|----------|----------|----------|-------------------|
All files    |      100 |    95.65 |      100 |      100 |                   |
 quadtree.js |      100 |    95.65 |      100 |      100 |           136,168 |
-------------|----------|----------|----------|----------|-------------------|
```
